### PR TITLE
Added sparse_factors function

### PR DESCRIPTION
**Pull request by sglyon**
Hi @timholy this is the spec builder function from over at Interpolations.jl

I've opened a PR to give you a chance to give feedback, especially regarding the name and whether or not to export the function.

Thanks!

### CODE REVIEW

File: src/WoodburyMatrices.jl [hunk 1]
```diff
@@ -98,4 +98,36 @@ function A_ldiv_B!(W::Woodbury, B::AbstractVector)
...
+```julia
+out = zeros(n, n)
+
+for (i, j, v) in args
+    out[i, j] = v
+end
+```
+
+"""
+function _build_woodbury_specs{T}(::Type{T}, n::Int, args::Tuple{Int, Int, Any}...)
```
Comment (timholy) [inline src/WoodburyMatrices.jl]: The code is fine---it's not high-performance, but this function doesn't need to be, if we don't export it, because you're using it only for small problems.

The help text is a little hard to follow: what is `args`? (it's clear if you read the code, but...). You might want to mention that `rowspec`, `valspec`, and `colspec` are all sparse matrices.

**Comment by timholy** [comment]
If we don't export it, the name isn't crucial, but somehow the name seems like it could be a little better. `woodbury` probably doesn't need to be in the name anymore, since this is what the package is about. What does the `specs` part of the name mean?

Maybe call it `sparse_factors`, in honor of the `sparse` function and the output?

**Comment by sglyon** [comment]
Thanks for the comments. I definitely agree that `woodbury` should be taken out of the name.

The `specs` part was just inherited from the Interpolations code where we were calling these matrices `rowspec`, `colspec`, `valspec`.

I like the `factor` idea much better.  Right now the `valspec` matrix is dense, but it really just needs to be a `Diagonal`. If we made that change, would it make sense to use `sparse_factors` still? I think so as diagonal is just a special case of sparse in my mind

**Comment by timholy** [comment]
Sure, that's still sparse, even if not a `SparseMatrixCSC` (which is an implementation detail, in my opinion).

**Comment by sglyon** [comment]
So I tried allowing the `vals` part be a `Diagonal`, but there is no method `*(::SparseMatrixCSC, ::Diagonal)` (ref JuliaLang/julia#14416)

For now it is also just a `SparseMatrixCSC`

**Comment by timholy** [comment]
Sorry I didn't notice this was ready. Thanks!

See also #6.

**Comment by sglyon** [comment]
No problem, thanks!

File: src/WoodburyMatrices.jl [hunk 2]
```diff
@@ -98,4 +98,54 @@ function A_ldiv_B!(W::Woodbury, B::AbstractVector)
...
+                                                 (1, 1, 2.0),
+                                                 (2, 2, 3.0),
+                                                 (3, 3, 4.0));
+
+julia> r*c*v - Diagonal([2.0, 3.0, 4.0])
+3x3 sparse matrix with 0 Float64 entries:
+```
+
+"""
+function sparse_factors{T}(::Type{T}, n::Int, args::Tuple{Int, Int, Any}...)
```
Comment (tkelman) [inline src/WoodburyMatrices.jl:131]: this needs Compat
